internal/app/secretless/handlers/pg/protocol/ssl.go

```diff
@@ -0,0 +1,149 @@
+package protocol
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"net"
+)
+
+type options map[string]string
+
+type TLSConfigWrapper struct {
+	UseTLS bool
+	tls.Config
+	VerifyCaOnly bool
+	Options options
+}
```

Comment:
I think, as a general rule, the name `XXXWrapper` indicates a problem.  It's telling you: I don't know really know what this thing _is_, so I'll name it after a thing that describes the way I'm _implementing_ it.

I _think_ in this case the missing concept is `SecretlessTLSConfig`....